src/ocaml/typing/403/oprint.ml

```diff
@@ -30,17 +30,19 @@ let rec print_ident ppf =
   | Oide_apply (id1, id2) ->
       fprintf ppf "%a(%a)" print_ident id1 print_ident id2
 
-let parenthesized_ident name =
-  (List.mem name ["or"; "mod"; "land"; "lor"; "lxor"; "lsl"; "lsr"; "asr"])
-  ||
-  (match name.[0] with
-      'a'..'z' | 'A'..'Z' | '\223'..'\246' | '\248'..'\255' | '_' ->
-        false
-    | _ -> true)
+let parenthesized_ident = function
+  | "or" | "mod" | "land" | "lor" | "lxor" | "lsl" | "lsr" | "asr" -> true
+  | "[]" | "()" -> false
+  | name ->
+    match name.[0] with
+    | 'a'..'z' | 'A'..'Z' | '\223'..'\246' | '\248'..'\255' | '_' -> false
+    | _ -> true
 
 let value_ident ppf name =
   if parenthesized_ident name then
-    fprintf ppf "( %s )" name
+    if name <> "" && (name.[0] = '*' || name.[String.length name - 1] = '*')
+    then fprintf ppf "( %s )" name
+    else fprintf ppf "(%s)" name
```

Comment:
Why do that extra work here (and in 4.02) but not in any of the following versions?

Reply:
Following versions handle that differently, but have the right behavior (otherwise the tests would fail).